packages/react-reconciler/src/__tests__/ReactNewContext-test.internal.js

```diff
@@ -1558,4 +1558,31 @@ Context fuzz tester error! Copy and paste the following line into the test suite
       }
     });
   });
+
+  it('should warn with an error message when using context as a consumer in DEV', () => {
+    const BarContext = React.createContext({value: 'bar-initial'});
+    const BarConsumer = BarContext;
+
+    function Component() {
+      return (
+        <React.Fragment>
+          <BarContext.Provider value={{value: 'bar-updated'}}>
+            <BarConsumer>
+              {({value}) => <div actual={value} expected="bar-updated" />}
+            </BarConsumer>
+          </BarContext.Provider>
+        </React.Fragment>
+      );
+    }
+
+    expect(() => {
+      ReactNoop.render(<Component />);
+      ReactNoop.flush();
+    }).toLowPriorityWarnDev(
+      'You are using the Context from React.createContext() as a consumer.' +
+        'The correct way is to use Context.Consumer as the consumer instead. ' +
```

Comment:
This message might confuse people a little bit. Maybe make it more visual?

```
Rendering <Context> directly is not supported and will be removed in a future major release. Did you mean to render <Context.Consumer> instead?
```

Reply:
This message is still not clear @trueadm . I'm already calling Context.Consumer as a consumer? But I'm still getting the error message. 

```jsx
import React from "react"

const defaultState = {
  dark: false,
  toggleDark: () => {},
}

const ThemeContext = Context.Consumer(defaultState)

// Getting dark mode information from OS!
// You need macOS Mojave + Safari Technology Preview Release 68 to test this currently.
const supportsDarkMode = () =>
  window.matchMedia("(prefers-color-scheme: dark)").matches === true

class ThemeProvider extends React.Component {
  state = {
    dark: false,
  }

  toggleDark = () => {
    let dark = !this.state.dark
    localStorage.setItem("dark", JSON.stringify(dark))
    this.setState({ dark })
  }

  componentDidMount() {
    // Getting dark mode value from localStorage!
    const lsDark = JSON.parse(localStorage.getItem("dark"))
    if (lsDark) {
      this.setState({ dark: lsDark })
    } else if (supportsDarkMode()) {
      this.setState({ dark: true })
    }
  }

  render() {
    const { children } = this.props
    const { dark } = this.state
    return (
      <ThemeContext.Provider
        value={{
          dark,
          toggleDark: this.toggleDark,
        }}
      >
        {children}
      </ThemeContext.Provider>
    )
  }
}

export default ThemeContext

export { ThemeProvider }
```

Reply:
@laurosilvacom you should be using ThemeContext = createContext(...) and then, doing ThemeContext.Consumer and ThemeContext.Provider

Reply:
I'm not calling `const ThemeContext = React.createContext(defaultState)` but still getting the same error. Am I missing something about how I'm suppose to call `<Context.Consumer>` instead?

Reply:
I was suggesting you should call it and use it that way. I can help more when I’m back in the office in a few days.

Reply:
Sounds good. I appreciate the help.

Reply:
@laurosilvacom 

I don't understand what you're trying to do.

You should create context like this:

```js
const ThemeContext = React.createContext(defaultState) // NOT "Context.Consumer(defaultState)"
```

and then subscribe to it either like this:

```js
class Button extends Component {
  static contextType = ThemeContext; // NOT ThemeContext.Consumer

  render() {
    // ...
  }
}
```

or like this:

```js
class Button extends Component {
  render() {
    return (
      // NOT <ThemeContext>
      <ThemeContext.Consumer>
        {theme => ...}
      </ThemeContext.Consumer>
    );
  }
}
```

Does that help?

Reply:
Yes!

I did not realize I needed to subscribe to it by calling ` <ThemeContext.Consumer>`.  

I had:

 <ThemeContext>
        {theme => ...}
 </ThemeContext>

Thanks @gaearon!

Reply:
Right, which is why the warning says:

>Rendering `<Context>` directly is not supported and will be removed in a future major release. Did you mean to render `<Context.Consumer>` instead?

Do you still feel it was unclear? Which part could be improved? Thanks!

Reply:
An error message that's direct would help. 

> Rendering <Context> directly is not supported and will be removed in a future major release. You want to use `Context.Consumer>`.